ProjectSidewalk/SidewalkWebpage
# Added label validation legend filter and popup to user dashboard map and admin view of user dashboard

Resolves #2811

Added validation filter to user profile map in legend. Added popup to user labels on map - removed validation options since user shouldn't be able to validate own labels. Reduced vertical spacing in legend to fit page. Needed to modify API which fetched data for user map by adding more fields in order to use filterLabelLayers function.

##### Before Screenshots
<img width="260" alt="image" src="https://github.com/user-attachments/assets/169e1603-b30b-4a6d-8dd4-e0ecc289605c">

<img width="896" alt="image" src="https://github.com/user-attachments/assets/b8c30ced-e744-470e-8c19-605617b6cff9">

<img width="1512" alt="Pasted Graphic 2" src="https://github.com/user-attachments/assets/586eb4e1-4c80-47ad-b0db-43aecb25ade6">

##### After Screenshots
<img width="260" alt="image" src="https://github.com/user-attachments/assets/2a45488d-0d1a-43f9-a661-806deb416238">

<img width="896" alt="image" src="https://github.com/user-attachments/assets/b3c8cd43-a28c-47e6-b287-03bbf3c7fc71">

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/c6361d7a-0fab-4cfd-bf4b-62fc297b733d">

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/8fa13340-ad46-4cf9-b9b9-1761bb9e7a51">

##### Testing instructions
1. Start local instance
2. Go to 'Your dashboard'
3. Check/uncheck legend filters and see that labels on map reflect state of filters
4. Click on popups for labels on the map

##### Things to check before submitting the PR
